tests/std/tests/P0896R4_common_iterator/test.cpp

```diff
@@ -226,6 +226,10 @@ using ICID = iterator_traits<common_iterator<input_copy_but_no_eq, default_senti
 STATIC_ASSERT(same_as<typename ICID::iterator_category, input_iterator_tag>);
 
 struct poor_sentinel {
+    poor_sentinel() = default;
+    poor_sentinel(const poor_sentinel&) {} // non-trivial copy constructor, to test _Variantish behavior

```

Comment:
I am not sure how the test passed!

Reply:
> I am not sure how the test passed!

(Actually looks at test) Oh, we don't ever copy a `common_iterator<int*, poor_sentinel>` that holds a `poor_sentinel`, we only copy a `common_iterator<int*, poor_sentinel>` that holds an `int*`. That _should_ work regardless of this copy constructor being `constexpr` or `trivial` since it isn't used. Of course this also suggests that the test isn't testing what we want it to, namely that a `common_iterator` holding a value with non-trivial but `constexpr` copy construction can be copied in the evaluation of a constant expression.

Reply:
Thanks, makes sense now! It was actually properly testing the missing `constexpr` on the constrained `_Variantish` copy constructor, since that constructor is selected if the sentinel isn't trivially copy constructible (even if the actual evaluation doesn't end up copying the sentinel). The test could probably be enhanced further but I think we're good for this PR.

Reply:
Too late! I pushed another change to get coverage of copying a stored value with non-trivial copy construction.